### PR TITLE
Move process action to new configuration pattern

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR implements the new action configuration logic introduced in https://github.com/dust-tt/dust/pull/5215 for the process action. I had to make some adjustments to the TS runners getter to fully support Partial (this time!).

## Risk

Worst case, it breaks the process action. Safe to rollback.

### CODE REVIEW

File: front/lib/api/assistant/actions/process.ts [hunk 1]
```diff
@@ -33,6 +33,8 @@ import {
   retrievalAutoTimeFrameInputSpecification,
```
Comment (flvndvd) [inline front/lib/api/assistant/actions/process.ts:33]: Better review with hide whitespace (click [here](https://github.com/dust-tt/dust/pull/5244/files?diff=split&w=1)!).

File: front/lib/api/assistant/actions/dust_app_run.ts [hunk 2]
```diff
@@ -121,10 +121,7 @@ export class DustAppRunConfigurationServerRunner extends BaseActionConfiguration
   // Generates the action specification for generation of rawInputs passed to `run`.
   async buildSpecification(
     auth: Authenticator,
-    {
-      name,
-      description,
-    }: { name?: string | undefined; description?: string | undefined }
+    { name, description }: { name?: string; description?: string }
```
Comment (philipperolet) [inline front/lib/api/assistant/actions/dust_app_run.ts:124]: aah I was wondering why the undefineds :)

File: front/lib/api/assistant/actions/process.ts [hunk 3]
```diff
@@ -142,383 +144,376 @@ export class ProcessAction extends BaseAction {
  * Params generation.
  */
 
-async function processActionSpecification({
```
Comment (philipperolet) [inline front/lib/api/assistant/actions/process.ts:145]: I'm assuming this function was moved without changes below in the file

File: front/lib/api/assistant/actions/process.ts [hunk 4]
```diff
@@ -142,383 +144,376 @@ export class ProcessAction extends BaseAction {
...
 
-/**
- * Action rendering.
- */
+    const { actionConfiguration } = this;
 
-// Internal interface for the retrieval and rendering of a actions from AgentMessage ModelIds. This
-// should not be used outside of api/assistant. We allow a ModelId interface here because for
-// optimization purposes to avoid duplicating DB requests while having clear action specific code.
-export async function processActionTypesFromAgentMessageIds(
```
Comment (philipperolet) [inline front/lib/api/assistant/actions/process.ts:203]: I'm assuming this was moved without changes below in the file

**Review by philipperolet — APPROVED** [review]
LGTM :fire:   trusting you on the types, and on the fact that you tested the "Extract" action to see if the refactor worked :+1:

File: front/lib/api/assistant/actions/process.ts (hunk 4 above)
Comment (flvndvd) [reply front/lib/api/assistant/actions/process.ts:203]: Correct

**Pull request merged**